server/src/main/java/org/elasticsearch/search/query/QueryPhase.java

```diff
@@ -303,6 +345,74 @@ static boolean execute(SearchContext searchContext,
         }
     }
 
+     private static Query tryRewriteNumericLongSort(SearchContext searchContext, IndexReader reader) throws IOException {
+        // check sort
+        Sort sort = searchContext.sort().sort;
+        if (Sort.RELEVANCE.equals(sort)) return null;
+        if (Sort.INDEXORDER.equals(sort)) return null;
+        if (sort.getSort().length > 1) return null; // we need only a single sort
```

Comment:
I don't think this should prevent the optimization to kick in. The only condition is that the primary sort is performed on a numeric field indexed with points and doc_values no matter what fields is used for tiebreaking.